### ISSUE TITLE
DomPDF won't render simple HTML page with 2 big images

### ISSUE THREAD

**Issue by manuel-funk** [dompdf/dompdf#2084]
The page https://www.rittal.com/pdf/test.html has two jpgs with a filesize of abou 2 mb each.
When I try to render it, I get the error message "C:\Users\%USER%\AppData\Local\Temp\xyz.pdf.part could not be saved because the source file could not be read. Try again later or contact the server administrator."
If I only use one of the 2 images in the source file it works.
What can I do to render a page with 2 or even more of thes images?

The page https://www.rittal.com/pdf/pdf.php contains the code to render the page.

And this is the code:
```
<html>

<head>
<title>Rittal-PDF</title>
</head>



<?php

$site     = "https://www.rittal.com/pdf/test.html";
$filename = "test.pdf";	

require_once 'Dompdf\autoload.inc.php';

use Dompdf\Dompdf;
use Dompdf\Options;

$options = new Options();
$options->set('isRemoteEnabled', TRUE);
$dompdf = new Dompdf($options);
$dompdf->load_html_file($site);
$dompdf->setPaper('A4', 'portrait');
$dompdf->render();
$dompdf->stream($filename);

?>


</html>
```

**Comment by bsweeney** [comment]
First thing you need to do it remove the HTML elements around your PHP. That's breaking everything by mixing HTML into the PDF stream. You code should just be:

```
<?php

$site     = "https://www.rittal.com/pdf/test.html";
$filename = "test.pdf";	

require_once 'Dompdf\autoload.inc.php';

use Dompdf\Dompdf;
use Dompdf\Options;

$options = new Options();
$options->set('isRemoteEnabled', TRUE);
$dompdf = new Dompdf($options);
$dompdf->load_html_file($site);
$dompdf->setPaper('A4', 'portrait');
$dompdf->render();
$dompdf->stream($filename);
?>
```

Second, when I try to load test.html the request for the linked stylesheet (`https://rinet.rittal.de/includes/style.css`) times out. That could be interfering with the rendering.

**Comment by manuel-funk** [comment]
Thanks for the quick reply!

I removed the HTML elements from https://www.rittal.com/pdf/pdf.php and also the linked stylesheet from https://rinet.rittal.de/includes/style.css.
When I then opened pdf.php it worked one time. Upon reload I got the same error as mentioned above.
So I tried it many times now. In most cases I get the error, but a few times it works.
Any idea what could be the problem?

**Comment by bsweeney** [comment]
Yeah, I am seeing the same thing. I think this may be an issue with the communication optimizations made around HTTP/2.

Not sure if the issue is on the server or the client, but I think this might be a Firefox-specific issue.

I tried this out on two systems using Firefox 72.0.2, one Windows one OS X. With HTTP/2 the first request succeeds but follow-up requests receive the error. The network panel in the dev console doesn't really indicate what's happening since it doesn't support HTTP/2 (see [this comment on bugzilla](https://bugzilla.mozilla.org/show_bug.cgi?id=1427256#c3)). I tried using Fiddler to see if there's some kind of communication issue but it doesn't support HTTP/2.

So then I tried Chrome and everything was fine :/

I believe (but could not find any info anywhere) the reason you see this intermittently is that if you wait long enough the HTTP/2 connection will close and the next request will start fresh. Not sure where to go with this one. You might need to do some investigation of your server settings. Or maybe even submit a bug report with Mozilla.

I'm going to keep this open for reference but right now I'm not sure there's anything we can do about it. Maybe if you tweak your script a bit? One issue we've seen in the past is the connection not really closing after dompdf streams the PDF, maybe a similar problem here since the connection is kept alive for HTTP/2. Try adding an `exit()` statement at the end of your script.

**Comment by goldfoot** [comment]
I am still seeing this behavior with files generated by dompdf v0.85 in Firefox 88.0.1 using HTTP/2. Works fine with HTTP/1.1, though. Adding `exit()` to the script solved the issue in my case.
